### ISSUE TITLE
Geth package ancient folder

### ISSUE THREAD

**Issue by luisalrp** [dappnode/DAppNode#250]
Full blockchain need. lot of storage that most of it does not have sense to have it in a fast and NVMe or SSD, because it not accessed too much.

geth already has"--datadir.ancient" parameter to store ancient data, so I would like that the installer of geth package asks where to install the ancient data instead of the whole package. That way, I can mount  cheap HDD nd point ancient data there. I think both options does not have sense.

Thanks!

**Comment by dapplion** [comment]
@luisalrp I really like the idea, thanks for the suggestion. @eduadiez what do you think?

**Comment by dapplion** [comment]
@eduadiez bump

**Comment by JHGrove3** [comment]
This sounds great!

Geth added the ability to create a Freezer on a hard drive.  Check out this blog post -- see the section titled "Freezer"

https://blog.ethereum.org/2019/07/10/geth-v1-9-0/

**Comment by Michael-A-Heuer** [comment]
Yes, please support an ancient drive. That would be really a great feature.

**Comment by pablomendezroyo** [comment]
@3alpha Would it be hard to implement this? I guess it affects mainly dappmanager installation process

**Comment by Michael-A-Heuer** [comment]
Not only for geth, but also for [Lighthouse's freezer DB feature](https://lighthouse-book.sigmaprime.io/advanced-datadir.html) (https://github.com/sigp/lighthouse/pull/508#issuecomment-558465291)
```
--freezer-dir <DIR>  Data directory for the freezer database.
```
This has a lot of advantages:
- It would give people more room to run an Ethereum full node with just 1 TB SSD space by using much cheaper HDD space.
- It make it easier to resync when you
    - you migrate to another machine as you don't need to download the entire state again or
    - your hot DB got corrupted because of a crash

**Comment by 3alpha** [comment]
> @3alpha Would it be hard to implement this? I guess it affects mainly dappmanager installation process

I don't think so. I think it could be dealt with within Installation wizard. It would only become messy if we were to deal with migrations. At first, I wouldn't deal with that. Maybe down the road.

**Comment by Michael-A-Heuer** [comment]
Any update on this? Would be really nice to have 🙏

**Comment by Michael-A-Heuer** [comment]
Since the disk requirement will increase after the Dencun upgrade (only [~50 GB](https://consensys.io/blog/ethereum-evolved-dencun-upgrade-part-5-eip-4844) but eventually more in the future), this becomes more of a topic.

**Comment by Michael-A-Heuer** [comment]
A freezer / ancient DB is also supported by Reth now.

See the idea https://github.com/paradigmxyz/reth/discussions/6510 and actual PR https://github.com/paradigmxyz/reth/pull/6444.

**Comment by Szboy** [comment]
I was able to get geth to connect to an external ancient folder by binding the default ancient location to my external hdd path however a native solution in DAppNode is very much needed, especially with the size of the chain now, and would save a lot of headaches.
